### ISSUE TITLE
Container (and Docker) keeps restarting

### ISSUE THREAD

**Issue by vivekkodira1** [docker/for-mac#186]
### Expected behavior
Container/Docker application should not restart
### Actual behavior
Container and Docker application keep restarting
### Information
Both the containers setup and the application itself keep restarting for no visible reason. 

Diagnostic ID: F7427C3F-5859-4184-94CF-21A364CABA1D
Docker for Mac: 1.12.0 (Build 10871)
macOS: Version 10.11.6 (Build 15G31)
[OK]     docker-cli
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce
Not applicable.

**Comment by ddebasilio** [comment]
I'm currently experiencing this too. I recently switched from the toolkit to native docker for mac and the application randomly restart which means I have to be bringing my containers up all the time. It will sometimes be stable for hours and sometimes just for minutes.

**Comment by samoht** [comment]
Do you still see that issue on more recent version of Docker for Mac?

**Comment by ddebasilio** [comment]
Yes, docker crashes constantly on Friday I even had a cron job checking the status to bring the containers up automatically because it was crashing every minute. 
I am on :
Docker version 1.12.1, build 6f9534c

**Comment by samoht** [comment]
@ddebasilio That's very unfortunate :-( Does it crash even when you do nothing? Could you try producing a diagnostic ID after the next crash so we could investigate? Thanks.

**Comment by ddebasilio** [comment]
It happens even when I do nothing. I can run docker-compose up -d and if I leave it idle it will eventually restart docker. 

$ /Applications/Docker.app/Contents/Resources/bin/docker-diagnose
OS X: version 10.12 (build: 16A323)
Docker.app: version: 1.12.1 (2d5b4d9)
Local time: Mon 10 Oct 2016 12:49:44 CEST
UTC:        Mon 10 Oct 2016 10:49:44 UTC
Timestamp:  20161010-124944
Running diagnostic tests:
[OK]      docker-cli
[OK]      Moby booted
[OK]      driver.amd64-linux
[OK]      vmnetd
[OK]      osxfs
[OK]      db
[OK]      slirp
[OK]      disk
[OK]      menubar
[ERROR]   environment      the variable DOCKERFILE_SUFFIX should not be set
[OK]      Docker
[OK]      VT-x
[OK]      kern.hv_support
Most specific failure is: the variable DOCKERFILE_SUFFIX should not be set
Docker logs are being collected into /tmp/51AFC3B8-3A8E-446E-B52D-2346BDFAF8B0/20161010-124944.tar.gz
Would you like to upload log files? [Y/n]: y

Your unique id is: 51AFC3B8-3A8E-446E-B52D-2346BDFAF8B0
Please quote this in all correspondence.

**Comment by samoht** [comment]
@ddebasilio Thanks! Your logs contains:

```
Oct 10 12:49:35 mac-264-ddb Docker[com.docker.driver.amd64-linux][9962] <Notice>: Assertion failed: (!REPLY_RING_EMPTY(sc)), function send_response_common, file src/pci_virtio_sock.
c, line 950.
```

Which seems to indicate that your issue is the same as https://github.com/docker/hyperkit/issues/50 and will be fixed in Beta28 (which will be released in the next few days). Would be great if you could test and report if your restart issue still persist.

**Comment by samoht** [comment]
@vivekkodira1 your initial report contains:

```
> Aug  8 11:37:41 vivek Docker[com.docker.driver.amd64-linux][1476] <Notice>: VM: 2016-08-08 06:07:41 +0000 UTC daemon.info vsudd: 124 Done. read: 96 written: 4249
> Aug  8 11:37:49 vivek Docker[com.docker.driver.amd64-linux][1476] <Notice>: virtio-net-vpnkit: initialising, opts="uuid=6122602e-37ed-4d3d-b0fb-7ea4a4c27a81,path=/Users/vivekkodira/Library/Containers/com.docker.docker/Data/s50,macfile=/Users/vivekkodira/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/mac.0"
> Aug  8 11:37:49 vivek Docker[com.docker.driver.amd64-linux][1476] <Notice>: Assertion failed: (sc->reply_cons != sc->reply_prod), function send_response_common, file src/pci_virtio_sock.c, line 879.
> Aug  8 11:37:49 vivek Docker[com.docker.driver.amd64-linux][1476] <Notice>: Interface will have uuid 6122602e-37ed-4d3d-b0fb-7ea4a4c27a81
> Aug  8 11:37:49 vivek Docker[com.docker.driver.amd64-linux][1476] <Notice>: Connection established with MAC=c0:ff:ee:c0:ff:ee and MTU 1500
> Aug  8 11:37:49 vivek Docker[com.docker.driver.amd64-linux][1476] <Notice>: virtio-9p: initialising path=/Users/vivekkodira/Library/Containers/com.docker.docker/Data/s40,tag=db✔ 1
```

which seems to be related to https://github.com/docker/hyperkit/issues/50 too. As the fix will be release in the next few days, I am closing the issue pro-actively. Please re-open it if the issue persists after Beta 28. /cc @ijc25

**Comment by ddebasilio** [comment]
Thanks! This would be awesome. I will see if I can install the Beta for testing but since I am using a laptop from work I may be busy with developing.

**Comment by ddebasilio** [comment]
So I still am having issues with docker crashing on my mac constantly. I have been able to install the Beta version for testing. 
  Docker version 1.12.3, build 6b644ec, experimental (beta29.2)
My issue should have been resolved with Beta28 but it seems that it still is constantly crashing/restarting. Once more I'm happy to provide any feedback that will help resolve this issue since I can barely work with docker right now.
unique id: 47ABEFF9-D769-481F-B8B1-279C5C1664FE
